services/scanners/dns/dns_scanner.py

```diff
@@ -34,19 +35,49 @@ async def scan_dmarc(domain):
 
     logging.info("Initiating DMARC scan")
 
-    # Single-item list to pass off to check_domains function
+    # Single-item list to pass off to check_domains function.
     domain_list = list()
     domain_list.append(domain)
 
     try:
-        # Perform "checkdmarc" scan on provided domain
+        # Perform "checkdmarc" scan on provided domain.
         scan_result = json.loads(json.dumps(check_domains(domain_list, skip_tls=True)))
     except (DNSException, SPFError, DMARCError) as e:
         logging.error(
-            "Failed to check the given domains for DMARC/SPF records: %s" % str(e)
+            "Failed to check the given domains for DMARC/SPF records. (%s)" % str(e)
         )
         return None
 
+    # Retrieve 'rua' tag address.
+    try:
+        rua_addr = scan_result['dmarc']['tags']['rua']['value'][0]['address']
+    except KeyError as e:
+        logging.error(
+            "Failed to retrieve RUA address from tag value. Report acceptance check will not be performed. (KeyError: %s)" % str(e)
+        )
+        rua_addr = None
+
+    if rua_addr is not None:
+        # Extract the domain from the address string (e.g. 'dmarc@cyber.gc.ca' -> 'cyber.gc.ca').
+        rua_domain = rua_addr.split("@", 1)[1]
+
+        # If the report destination differs from the originating domain, assert reports are being accepted.
+        if rua_domain == domain:
+            scan_result['dmarc']['tags']['rua']['accepting'] = True
```

Comment:
If `rua_domain` has the same organizational domain as `domain` (i.e. one level below the public suffix), we don't need to do the check. So, `tracker.cyber.gc.ca` sending to `cyber.gc.ca` should pass, since `gc.ca` is a public suffix.

There's a public suffix library at https://pypi.org/project/publicsuffixlist/